pkg/driver/provisioner.go

```
@@ -0,0 +1,182 @@
+/*
+ Copyright 2022 Juicedata Inc
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package driver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/juicedata/juicefs-csi-driver/pkg/juicefs"
+	k8s "github.com/juicedata/juicefs-csi-driver/pkg/juicefs/k8sclient"
+	"github.com/juicedata/juicefs-csi-driver/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+	"os"
+	provisioncontroller "sigs.k8s.io/sig-storage-lib-external-provisioner/v6/controller"
+	"strconv"
+)
+
+var (
+	provisionerSecretName      = "csi.storage.k8s.io/provisioner-secret-name"
+	provisionerSecretNamespace = "csi.storage.k8s.io/provisioner-secret-namespace"
+	publishSecretName          = "csi.storage.k8s.io/provisioner-secret-name"
+	publishSecretNamespace     = "csi.storage.k8s.io/provisioner-secret-namespace"
+)
+
+type provisionerService struct {
+	juicefs juicefs.Interface
+	*k8s.K8sClient
+}
+
+func newProvisionerService(k8sClient *k8s.K8sClient) (provisionerService, error) {
+	jfs, err := juicefs.NewJfsProvider(nil)
+	if err != nil {
+		panic(err)
```

why not return the error?